src/AutoRest.CSharp/Output/Models/Types/OutputLibrary.cs

```diff
@@ -170,5 +177,95 @@ public RestClient FindRestClient(OperationGroup operationGroup)
             EnsureHeaderModels().TryGetValue(operation, out var model);
             return model;
         }
+
+        private void DecorateOperationGroup()
+        {
+            foreach (var operations in _codeModel.OperationGroups)
+            {
+                operations.ProviderName = _context.Configuration.OperationGroupMapping.ContainsKey(operations.Key) ? _context.Configuration.OperationGroupMapping[operations.Key] : ConstructOperationProviderName(operations);
```

Comment:
Typically we don't modify the CodeModel during the generation. If some logic needs to be applied to the output model it needs to go into the corresponding output model class (https://github.com/Azure/autorest.csharp/blob/feature/v3/src/AutoRest.CSharp/Output/Models/Client.cs , https://github.com/Azure/autorest.csharp/blob/feature/v3/src/AutoRest.CSharp/Output/Models/RestClient.cs , https://github.com/Azure/autorest.csharp/blob/feature/v3/src/AutoRest.CSharp/Output/Models/Types/ObjectType.cs , etc.)